packages/components/src/alignment-matrix-control/utils.tsx

```diff
@@ -55,12 +69,15 @@ export function getItemId( prefixId, value ) {
 /**
  * Retrieves the alignment index from a value.
  *
- * @param {string} alignment Value to check.
+ * @param  alignment Value to check.
  *
- * @return {number} The index of a matching alignment.
+ * @return The index of a matching alignment.
  */
-export function getAlignmentIndex( alignment = 'center' ) {
+export function getAlignmentIndex(
+	alignment: AlignmentMatrixControlValue = 'center'
+) {
 	const item = transformValue( alignment ).replace( '-', ' ' );
+	if ( ! checkIsValidAlignment( item ) ) return;
```

Comment:
This new type guard (hat tip @ciampo) will avoid running `indexOf` on an invalid alignment value, but shouldn't introduce any real changes at runtime, as `getAlignmentValue` would have returned `undefined` anyway.

Reply:
If I'm reading this correctly, it looks like we were running a redundant `.replace( '-', ' ' )` here, since that's already done in the `transformValue()` function. Removing that should remove the need for an added type guard.

```diff
-	const item = transformValue( alignment ).replace( '-', ' ' );
+	const item = transformValue( alignment );
-	if ( ! checkIsValidAlignment( item ) ) return;
```

Reply:
That's correct. When pairing on it, Chad and I weighed our options and decided to add the type guard for the least amount of runtime changes.

That's because replace()` only replaces the first occurrence of the match, and so running `replace()` twice could actually have a different effect than running it once 🤷  

But I see your point — I'd be ok to apply your suggestions. If this change causes any breakages, it's likely because the `alignment` value being used was not following the specs anyway

Reply:
Sounds good to me, I've removed the new type guard and the second `.replace` call.